src/main/java/com/practice/controller/UserController.java

```diff
@@ -0,0 +1,54 @@
+package com.practice.controller;
+
+import com.practice.User;
+import com.practice.service.UserService;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.ModelAndView;
+
+import java.util.List;
+
+@Controller
+@RequestMapping("/user")
+public class UserController {
+
+    private UserService userService;
+
+    public UserController(UserService userService) {
+        super();
+        this.userService = userService;
+    }
+
+    @GetMapping
+    public String getAllUsers(Model model, User user) {
+         model.addAttribute("users", userService.getAllUsers());
+         model.addAttribute("user", user);
+         return "user";
+    }
+
+//    @GetMapping("/{id}")
+//    public String getUser(@PathVariable("id") int id) {
+//        userService.getUserById(id);
+//        return "user";
+//    }
+
+    @PostMapping
+    public String addUser(@ModelAttribute("user") User user) {
+        userService.addUser(user.getName(), user.getEmail(), user.getPhone());
```

Comment:
DO NOT pass individual parameters into service.

Think Object-oriented. So we need to pass objects from controller to service, not individual fields.